server/src/Irony.cpp

```diff
@@ -286,6 +286,50 @@ void Irony::complete(const std::string &file,
   std::cout << "(success . t)\n";
 }
 
+namespace {
+
+static bool hasUppercase(const std::string &prefix)
```

Comment:
You can remove `static` here and the other functions below, I prefer unnamed namespaces to static function.
This is mostly a matter of style and not a real issue, see also https://stackoverflow.com/a/4977525/951426